src/sdl2/render.rs

```diff
@@ -492,30 +496,81 @@ impl TextureCreator {
     }
 }
 
+pub enum TargetKind {
+    Canvas,
+    Texture,
+}
+
+pub trait Target {
```

Comment:
I feel like if this trait is only used for internal use, we might as well make it private for now. The user will never have to implement Target himself, so if we can prevent him from doing so, the better.